pymc_marketing/clv/models/basic.py

```diff
@@ -78,19 +97,125 @@ def _fit_MAP(self, **kwargs):
+        >>> imported_model = MyModel.load(name)
+        """
+        filepath = Path(str(fname))
+        idata = az.from_netcdf(filepath)
+        dataset = idata.fit_data.to_dataframe()
+
+        model = cls(
+            dataset,
+            model_config=json.loads(idata.attrs["model_config"]),  # type: ignore
+            sampler_config=json.loads(idata.attrs["sampler_config"]),
+        )
+        model.idata = idata
+
+        model.build_model()
+        # All previously used data is in idata.
+
+        if model.id != idata.attrs["id"]:
+            raise ValueError(
+                f"The file '{fname}' does not contain an inference data of the same model or configuration as '{cls._model_type}'"
+            )
+
+        return model
+
+    @staticmethod
+    def _check_prior_ndim(prior, ndim=0):
+        if prior.ndim != ndim:
+            raise ValueError(
+                f"Prior variable {prior} must be have {ndim} ndims, but it has {prior.ndim} ndims."
+            )
+
+    def create_distribution_from_prior(self, name: str, **kwargs) -> TensorVariable:
+        if name == "gamma":
+            return pm.Gamma.dist(**kwargs)
+        if name == "halfflat":
+            return pm.HalfFlat.dist(**kwargs)
+        if name == "flat":
+            return pm.Flat.dist(**kwargs)
+        if name == "normal":
+            return pm.Normal.dist(**kwargs)
+        if name == "halfnormal":
+            return pm.HalfNormal.dist(**kwargs)
+        if name == "halfcauchy":
+            return pm.HalfCauchy.dist(**kwargs)
+        if name == "halfstudentt":
+            return pm.HalfStudentT.dist(**kwargs)
+        if name == "normal":
+            return pm.Normal.dist(**kwargs)
+        if name == "diracdelta":
+            return pm.DiracDelta.dist(**kwargs)
+        else:
+            raise ValueError(f"Prior distribution {name} not supported")
+
+    @staticmethod
+    def _process_priors(
+        *priors: TensorVariable, check_ndim: bool = True
+    ) -> Tuple[TensorVariable, ...]:
+        """Check that each prior variable is unique and attach `str_repr` method."""
+        if len(priors) != len(set(priors)):
+            raise ValueError("Prior variables must be unique")
+        # Related to https://github.com/pymc-devs/pymc/issues/6311
+        for prior in priors:
+            if check_ndim:
+                CLVModel._check_prior_ndim(prior)
+            prior.str_repr = types.MethodType(str_for_dist, prior)  # type: ignore
+        return priors
 
     @property
-    def fit_result(self):
-        if self._fit_result is None:
+    def prior_predictive(self) -> az.InferenceData:
+        if self.idata is None or "prior_predictive" not in self.idata:
             raise RuntimeError("The model hasn't been fit yet, call .fit() first")
-        return self._fit_result
+        return self.idata["prior_predictive"]
+
+    @property
+    # if we include fit_result then this should be added for consistancy
```

Comment:
I can't quite tell what this comment means.

Reply:
Ah, yes prior predictive is less of a thing in CLV models in general

Reply:
But we can definitely add some property for it

Reply:
I think we should add the prior predictive property, because I have an idea on how it can be added for the `ParetoNBD` and `BetaGeoBetaBinom` models and even be used for data drift monitoring after this PR is merged and save/load functionality is supported.

Reply:
ok, So I guess the discussion on this one can be closed, I'll remove the comments and keep the prior_predictive and posterior_predictive properties to have the APIs unified

Reply:
sounds good